src/node/db/PadManager.js

```diff
@@ -49,17 +49,18 @@ var globalPads = {
  * Updated without db access as new pads are created/old ones removed.
  */
 let padList = {
-  list: [],
+  list: new Set(),
+  cachedList: [],
   sorted : false,
   initiated: false,
   init: async function() {
-    let dbData = await db.findKeys("pad:*", "*:*:*");
+    let dbData = await db.findKeys("pad:*", "pad:*:*");
```

Comment:
Nice potential **performance gain**.

If possible confirm my analysis, based on looking at the [mysql](https://github.com/ether/ueberDB/blob/5fbc4f3ac182956168860a6a27d11ca82e820836/mysql_db.js) driver (I did not look at other DB drivers).

Am I right that this results in going from this query:

```sql
SELECT key FROM store WHERE key LIKE 'pad:%' AND key NOT LIKE '%:%:%'
```

To this one?
```sql
SELECT key FROM store WHERE key LIKE 'pad:%' AND key NOT LIKE 'pad:%:%'
```

Looking at the [database documentation](https://etherpad.org/doc/v1.7.5/#index_pad_padid) the structure of DB keys is the following:
```
pad:$PADID
pad:$PADID:revs:$REVNUM
pad:$PADID:chat:$CHATNUM
```

So the purpose would be excluding any key that has a colon (`:`) after "pad:$PADID".

This change can be **split** in a dedicated commit and maybe we can add a comment before:

```js
/*
 * select only the keys of the form pad:$PADID, defensively excluding all the
 * others (for example pad:$PADID:revs:$REVNUM and pad:$PADID:chat:$CHATNUM)
 */
```


**Edit**: upon further analysis, I fear **this would break compatibility with some drivers**. See for example Redis:

https://github.com/ether/ueberDB/blob/5fbc4f3ac182956168860a6a27d11ca82e820836/redis_db.js#L81

```js
callback(new customError("redis db currently only supports *:*:* as notKey","apierror"), null);
```

We would break Redis compatibility.

If we want to include this, we have to fix https://github.com/ether/ueberDB/blob/5fbc4f3ac182956168860a6a27d11ca82e820836/redis_db.js. It's an old driver, maybe Redis has evolved in the meantime. I do not know.

We **also should check the effects with the other DBs**.

Reply:
I'll remove it. The performance gain is not very interesting, and we don't really want to break things.